packages/cli-kit/src/environment/spin-cache.ts

```diff
@@ -0,0 +1,9 @@
+let cachedSpinFQDN: string
+
+export function getCachedSpinFqdn(): string | undefined {
+  return cachedSpinFQDN
+}
+
+export function setCachedSpinFqdn(fqdn: string) {
+  cachedSpinFQDN = fqdn
+}
```

Comment:
Adding state to modules in `@shopify/cli-kit` caused issues in the past because we can't prevent multiple copies of `@shopify/cli-kit` from being in the graph and with them duplicated states. I'd consider loading this information once at the service level and passing it down. Alternatively, we shell out to get the information. Have you measured how much time it takes that operation?
cc @isaacroldan